File: backend/src/models/Material.js
const mongoose = require('mongoose');

const CATEGORIES = [
  'MOBILE_PCB',
  'COPPER_WIRE',
  'COMPUTER_PCB',
  'BATTERY_LEAD',
  'FAN_MOTOR',
  'COMPRESSOR',
  'LCD_PANEL',
  'CRT_TUBE',
  'CRT',
  'PCB',
  'CABLE',
  'BATTERY',
  'MOTOR',
  'MAGNET_ASSEMBLY',
  'MIXED_PLASTIC',
  'OTHER',
];

const materialSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    category: { type: String, enum: CATEGORIES, required: true },
    subCategory: { type: String, default: '', trim: true },
    description: { type: String, default: '' },
    unit: { type: String, default: 'kg' },
    isActive: { type: Boolean, default: true },
    isDemo: { type: Boolean, default: false },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Material', materialSchema);
module.exports.CATEGORIES = CATEGORIES;

File: android-app/app/src/main/java/com/kabadiwalaconnect/data/model/Material.kt
package com.kabadiwalaconnect.data.model

data class Material(
    val id: String,
    val name: String,
    val category: MaterialCategory,
    val unit: String = "kg",
    val pricePerUnit: Double,
    val isActive: Boolean = true,
    val createdAt: String,
    val updatedAt: String
)

enum class MaterialCategory {
    PAPER,
    PLASTIC,
    METAL,
    E_WASTE,
    GLASS,
    CARDBOARD,
    OTHER
}
